nomad/structs/structs_test.go

```diff
@@ -6024,3 +5917,30 @@ func TestTaskGroup_validateScriptChecksInGroupServices(t *testing.T) {
 		require.Nil(t, mErrOK)
 	})
 }
+
+func requireMultierrorNotContaining(t *testing.T, err error, unexpected ...string) {
```

Comment:
These function doesn't seem to actually rely on `err` being a `multierror.Error`.

Given that it's called in one place only, how does it compare to using:

```go
require.NoError(t, err)
require.NotContains(t, err.Error(), tt.unexpectedError)
```

Also, I find not-contains assertions to be relatively weak, and would be concern this helper may encourage an anti-pattern.  The test may pass silently and mask an error for an expected reasons, e.g. the errors messages changed since the test got updated and the used comparison string is no longer significant, the error got wrapped and some message detailed got lost/ignored, the test starts failing because of broken test setup and the test no longer exercise the relevant code, etc. A test that makes a positive assertion on the expected error is a strong test IMO.

Reply:
That's a good point about it not taking advantage of the fact that it's a multierror.Error. The original implementation looked at the content of individual errors within the multierror; it probably makes sense to continue doing that. Otherwise, the multierror's formatter is being tested as well, rather than (I think) just the individual error contents.

The unexpected error thing is weird because in the only user of it, the test will always return a collection of unrelated errors, but it should not contain any that mention the thing under test, which is the CSIPluginConfig. Previously the test would fail if the returned error object did not have N-1 errors, where the positive case for that test had N errors. :face_with_head_bandage: 

Does that make sense?

I think the only alternative is not to have a "positive" case for CSIPluginConfig validation, as other validation errors will exist.